content/en/platform/corda/4.8/open-source/key-concepts-contracts.md

```diff
@@ -12,111 +12,97 @@ menu:
 
-**Contract validity** is defined as follows:
+Putting a contract on Corda gives it unique features:
+* It can't be changed, only replaced with an updated version.
+* Once executed, the results are irreversible.
 
-* Each transaction state specifies a *contract* type
-* A *contract* takes a transaction as input, and states whether the transaction is considered valid based on the
-contract’s rules
-* A transaction is only valid if the contract of **every input state** and **every output state** considers it to be
-valid
+## Smart contract languages
+Corda smart contracts must be written in [Kotlin](https://kotlinlang.org/) or [Java](https://www.java.com/en/).
 
-We can picture this situation as follows:
+## Contractual validity
 
-{{< figure alt="tx validation" width=80% zoom="/en/images/tx-validation.png" >}}
-The contract code has access to the full capabilities of the language,
-including:
+[Transactions](key-concepts-transactions.md) must be digitally signed by all required signers. However, even if a
+transaction gathers all the required signatures, it can't be executed unless it is also *contractually valid*. A transaction that is not contractually valid is not a valid proposal to update the ledger, and can never be committed to the ledger. This means that contracts can impose rules on the evolution of states over time that are independent of the willingness of the required signers to sign a given transaction.
+
+Each transaction [state](key-concepts-states.md) specifies a *contract type*. The contract specified takes the transaction as input, and determines if the transaction is valid based on the
+contract's internal rules. The contract must evaluate every input state and every output state.
 
-* Checking the number of inputs, outputs, commands, or attachments
-* Checking whether there is a time window or not
-* Checking the contents of any of these components
-* Looping constructs, variable assignment, function calls, helper methods, and so on
-* Grouping similar states to validate them as a group; for example, imposing a rule on the combined value of all the cash
-states
+{{< figure alt="tx validation" width=80% zoom="/en/images/tx-validation.png" >}}
+The contract code can:
 
-A transaction that is not contractually valid is not a valid proposal to update the ledger, and thus can never be
-committed to the ledger. In this way, contracts impose rules on the evolution of states over time that are
-independent of the willingness of the required signers to sign a given transaction.
+* Check the number of inputs, outputs, commands, or attachments.
+* Check for [time windows](key-concepts-time-windows.md).
+* Check the contents of all components.
+* Evaluate looping constructs, variable assignments, function calls, and helper methods, and other aspects of the transaction code.
```

Comment:
```suggestion
* Evaluate looping constructs, variable assignments, function calls, helper methods, and other aspects of the transaction code.
```

This 'and' doesn't really need to go but I think it works just fine/better if it does.